src/uu/numfmt/src/numfmt.rs

```diff
@@ -28,13 +30,22 @@ const AFTER_HELP: &str = help_section!("after help", "numfmt.md");
 const USAGE: &str = help_usage!("numfmt.md");
 
 fn handle_args<'a>(args: impl Iterator<Item = &'a str>, options: &NumfmtOptions) -> UResult<()> {
+    let mut has_failed_flag = false;
     for l in args {
-        match format_and_print(l, options) {
-            Ok(_) => Ok(()),
-            Err(e) => Err(NumfmtError::FormattingError(e.to_string())),
-        }?;
+        let format_result = format_and_handle_validation(l, options);
+        if format_result.is_err() {
+            match options.invalid {
+                InvalidModes::Abort => return format_result,
+                InvalidModes::Fail => {
+                    has_failed_flag = true;
```

Comment:
Have you seen [`set_exit_code`](https://docs.rs/uucore/0.0.16/uucore/error/fn.set_exit_code.html) in `uucore`? That might a be a cleaner way of dealing with this. Essentially, if you return `Ok` from `main` it will try to get the exit code set via `set_exit_code`, which is useful in these kinds of situations. It will also be set if you use [`show!`](https://docs.rs/uucore/0.0.16/uucore/macro.show.html) on a `UError`.

Reply:
Good advice! Using show and show_error simplified things a lot :+1: